scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/SingleTaskScheduler.java

```diff
@@ -19,22 +19,47 @@
 import com.amazonaws.blox.scheduling.state.ClusterSnapshot.ContainerInstance;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
-import java.util.Map.Entry;
 
 /**
  * Temporary scheduler implementation that just starts a single task on the first instance in the
  * snapshot.
+ *
+ * <p>TODO Remove this, it's only a demo scheduler
  */
 public class SingleTaskScheduler implements Scheduler {
 
   @Override
   public List<SchedulingAction> schedule(ClusterSnapshot snapshot, EnvironmentVersion environment) {
-    // Only run a Task on the first ContainerInstance in the snapshot:
-    for (Entry<String, ContainerInstance> entry : snapshot.getInstances().entrySet()) {
+    // Naively only run a Task on the first ContainerInstance in the snapshot:
+    ContainerInstance instance =
+        snapshot
+            .getInstances()
+            .stream()
+            .sorted(Comparator.comparing(ContainerInstance::getArn))
```

Comment:
do we need to sort? I suppose this is gonna be removed anyway

Reply:
yes. Just so we don't start the task multiple times on different instances just because the ordering is different. (I had that before, and ended up with a cluster full of random tasks)